sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -111,6 +114,7 @@ public static ClientPipeline Create(
         pipelineLength += options.BeforeTransportPolicies?.Length ?? 0;
 
         pipelineLength++; // for retry policy
```

Comment:
Should we just do `pipelineLength += 3` rather than having to run multiple statements, since we know how many things we're adding?

Reply:
I personally like the clarity of incrementing for each policy and keeping track of which is which.